test/e2e/rules_api_test.go

```diff
@@ -42,15 +42,17 @@ func TestRulesAPI_Fanout(t *testing.T) {
 		s.SharedDir(),
 		netName,
 		"prom1",
-		defaultPromConfig("ha", 0, "", filepath.Join(e2e.ContainerSharedDir, rulesSubDir, "*.yaml")),
+		// Only use the first rule since Prometheus doesn't support native `partial_response_strategy` field in yaml.
+		defaultPromConfig("ha", 0, "", filepath.Join(e2e.ContainerSharedDir, rulesSubDir, "rules-0.yaml")),
```

Comment:
how this work before? Prometheus did never support this natively 🤔 Is this because of strict unmarshalling being introduced?

Reply:
Also curious was error from issue reproducible without your fix on 2.25 version? 🤗

Reply:
Looks like it is caused by the strict unmarshalling change. Prometheus cannot start if `partial_response_strategy` is added.

> Also curious was error from issue reproducible without your fix on 2.25 version

Yes, it is not related to my change.

```
14:14:59 prometheus-prom1: level=info ts=2021-03-10T19:14:59.567Z caller=main.go:928 msg="Loading configuration file" filename=/shared/data/prometheus/prom1/prometheus.yml
14:14:59 prometheus-prom1: level=error ts=2021-03-10T19:14:59.568Z caller=manager.go:946 component="rule manager" msg="loading groups failed" err="/shared/rules/rules-1.yaml: yaml: unmarshal errors:\n  line 5: field partial_response_strategy not found in type rulefmt.RuleGroup"
```

Reply:
Any idea about how to fix this strict unmarshalling thing? This is needed as we need to upgrade E2E Prometheus to the latest version to support exemplar tests.